test/cpu.cpp

```diff
@@ -907,12 +919,232 @@ TEST_CASE("PLP", "[cpu][instruction]") {
+		testInstruction(Opcode::TXA, AddressingMode::Implied,
+			/*addExpectedBusAccesses=*/noopAddBusAccesses,
+			/*setup=*/[&](CPU& cpu) {
+				cpu.e = 0;
+				cpu.X.forceStoreFull(val);
+			},
+			/*test=*/[&](CPU& cpu) {
+				REQUIRE(cpu.A.load() == cpu.X.load());
+			}
+		);
+	}
+}
+
+TEST_CASE("TAY", "[cpu][instruction]") {
+	auto indexRegistersAre8Bit = GENERATE(false, true);
+	auto val = GENERATE_COPY(take(1, random<Address>(0, (indexRegistersAre8Bit ? 0xff : 0xffff) + 1)));
+		
+	DYNAMIC_SECTION((indexRegistersAre8Bit ? 8 : 16) << "-bit") {
+		testInstruction(Opcode::TAY, AddressingMode::Implied,
+			/*addExpectedBusAccesses=*/noopAddBusAccesses,
+			/*setup=*/[&](CPU& cpu) {
+				cpu.e = 0;
+				cpu.A.forceStoreFull(val);
+			},
+			/*test=*/[&](CPU& cpu) {
+				REQUIRE(cpu.Y.load() == cpu.A.load());
+			}
+		);
+	}
+}
+
+TEST_CASE("TYA", "[cpu][instruction]") {
+	auto memoryAndAccumulatorAre8Bit = GENERATE(false, true);
+	auto val = GENERATE_COPY(take(1, random<Address>(0, (memoryAndAccumulatorAre8Bit ? 0xff : 0xffff) + 1)));
+		
+	DYNAMIC_SECTION((memoryAndAccumulatorAre8Bit ? 8 : 16) << "-bit") {
+		testInstruction(Opcode::TYA, AddressingMode::Implied,
+			/*addExpectedBusAccesses=*/noopAddBusAccesses,
+			/*setup=*/[&](CPU& cpu) {
+				cpu.e = 0;
+				cpu.Y.forceStoreFull(val);
+			},
+			/*test=*/[&](CPU& cpu) {
+				REQUIRE(cpu.A.load() == cpu.Y.load());
+			}
+		);
+	}
+}
+
+TEST_CASE("TXY", "[cpu][instruction]") {
+	auto indexRegistersAre8Bit = GENERATE(false, true);
+	auto val = GENERATE_COPY(take(1, random<Address>(0, (indexRegistersAre8Bit ? 0xff : 0xffff) + 1)));
+			
+	DYNAMIC_SECTION((indexRegistersAre8Bit ? 8 : 16) << "-bit") {
+		testInstruction(Opcode::TXY, AddressingMode::Implied,
+			/*addExpectedBusAccesses=*/noopAddBusAccesses,
+			/*setup=*/[&](CPU& cpu) {
+				cpu.e = 0;
+				cpu.X.forceStoreFull(val);
+			},
+			/*test=*/[&](CPU& cpu) {
+				REQUIRE(cpu.Y.load() == cpu.X.load());
+			}
+		);
+	}
+}
+
+TEST_CASE("TYX", "[cpu][instruction]") {
+	auto indexRegistersAre8Bit = GENERATE(false, true);
+	auto val = GENERATE_COPY(take(1, random<Address>(0, (indexRegistersAre8Bit ? 0xff : 0xffff) + 1)));
+			
+	DYNAMIC_SECTION((indexRegistersAre8Bit ? 8 : 16) << "-bit") {
+		testInstruction(Opcode::TYX, AddressingMode::Implied,
+			/*addExpectedBusAccesses=*/noopAddBusAccesses,
+			/*setup=*/[&](CPU& cpu) {
+				cpu.e = 0;
+				cpu.Y.forceStoreFull(val);
+			},
+			/*test=*/[&](CPU& cpu) {
+				REQUIRE(cpu.X.load() == cpu.Y.load());
+			}
+		);
+	}
+}
+
+TEST_CASE("TSX", "[cpu][instruction]") {
+	auto indexRegistersAre8Bit = GENERATE(false, true);
+	auto val = GENERATE_COPY(take(1, random<Address>(0, (indexRegistersAre8Bit ? 0xff : 0xffff) + 1)));
+			
+	DYNAMIC_SECTION((indexRegistersAre8Bit ? 8 : 16) << "-bit") {
+		testInstruction(Opcode::TSX, AddressingMode::Implied,
+			/*addExpectedBusAccesses=*/noopAddBusAccesses,
+			/*setup=*/[&](CPU& cpu) {
+				cpu.e = 0;
+				cpu.SP = val;
```

Comment:
You forgot to set the `x` flag according to `indexRegistersAre8Bit`.
```suggestion
				cpu.SP = val;
				cpu.setFlag(CPU::flags::x, indexRegistersAre8Bit);
```